instructure/canvas-ios
# Rework course invitation fetch and display logic.

refs: MBL-16050
affects: Student, Teacher
release note: Fixed invitations from not yet active courses not showing up on the dashboard.

test plan:
- Create a course, set its start date to the future.
- Invite a student/teacher to this course.
- Start the student/teacher app.
- Course invitation from this future course should show up.
- Accept it.
- Invitation banner should remove itself animated.
--
- Repeat the same steps but with an already ongoing course.
- Notice that after accepting the invitation the dashboard refreshes and the new course shows up.

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td><img src="https://user-images.githubusercontent.com/72396990/169552640-92022c73-a995-45a1-99e0-cffbb70c2951.png"></td>
<td><img src="https://user-images.githubusercontent.com/72396990/169552724-2cf91dfb-220b-484a-a5be-b4391e3ebfe7.png"></td>
</tr>
</table>
